source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -340,7 +345,13 @@ Http::FilterDataStatus AWSLambdaFilter::decodeData(Buffer::Instance &data,
     has_body_ = true;
   }
 
-  aws_authenticator_.updatePayloadHash(data);
+  // If we are not transforming the request, then update the payload hash according to the incoming data
+  // If we are transforming the request, then we will update the payload hash after the transformation
+  if (!isRequestTransformationNeeded()) {
+    aws_authenticator_.updatePayloadHash(data);
+  } else {
+    request_body_.move(data);
```

Comment:
we should avoid doing this, as now envoy cannot manage the size of the buffer.

Reply:
Resolved this here: https://github.com/solo-io/envoy-gloo/pull/177/commits/8b3c855703350391da50ba5677b14c0c7775a6ce

Reply:
Additionally, updated the tests in response to these changes here: https://github.com/solo-io/envoy-gloo/pull/177/commits/b80e4c155907eab46508d2d49c0176415e7c4039